### ISSUE TITLE
Alternate DB Backend *Enhancement Request*

### ISSUE THREAD

**Issue by camarokris** [nZEDb/nZEDb#2253]
Would it be possible to write in support for MongoDB. I believe that MongoDB would offer performance as well as a smaller footprint on the storage device. This would be extremely helpful for those that are indexing several hundred or thousand groups. 

Edit: I did see some references to MongoDB in the code but it was not clear if uncommenting those references is all that would have to be done to make it work or if more work would be needed.

**Comment by Bart39** [comment]
Highly unlikely as that would require a huge amount of work but let's see what the other dev's think

**Comment by peternavr** [comment]
I don't think this is applicable. Mongo is better for large flat data.

**Comment by niel** [comment]
Yes, it is possible. However I wouldn't want to take it on before we have a much greater integration of our code into the lithium framework (which already has first class MongoDb support).
This is definitely something I would like to experiment with, to see if it would offer any kind of benefit.

**Comment by OliverLeitner** [comment]
i doubt that changing the database will fix anything. better to find the weaknesses in the current implementation.

**Comment by jangrewe** [comment]
Don't invest any time in it until you read this, especially the Epilogue: http://www.sarahmei.com/blog/2013/11/11/why-you-should-never-use-mongodb/

(I'm not saying it's a bad idea, i just think we're fine with e.g. MySQL and Sphix ;-) )

**Comment by b3rs3rk** [comment]
@jangrewe is right.  We have too many table relationships to even consider it.

**Comment by niel** [comment]
Agreed for the most part, but I do think CBP could be a good candidate, which would drastically increase speeds creating new releases and dropping the old data.
